### DISCUSSION TITLE
[Chapter 08] OLTP

### DISCUSSION THREAD

**Discussion by BvrPark** [cowaine/spring-boot-for-msa#83] [Chapter 08. Data Persistence]
1. p.457에 보면 OLTP와 OLAP에 대한 내용이 나옵니다.
OLTP는 트랜잭션 처리, OLAP는 통계에 쓰인다고 나오는데 OLTP, OLAP가 쿼리를 처리하는 한 방법인지 아니면 쿼리의 종류를 저렇게 나누는 건지 그것이 궁금합니다.

**Comment by corock** [comment]
온라인 트랜잭션 처리(_Online Transactional Processing_)와 온라인 분석 처리(_Online Analytical Processing_)는
쿼리를 처리하는 방법이나 쿼리의 종류에 따라 구분하는 것 같진 않는 것 같습니다.
책에 나온 대로 처리 목적에 맞게 구분하고, OOP 와 AOP 처럼 대비되는 개념이 아니라 상호 보완적인 관계가 아닐까요?

예를 들면, 회원 가입이나 주문서 생성이나 결제 승인 등 요구사항 개발을 위한 CRUD 가 다 적용되는 것이 OLTP 이고,
이번 달 우리 쇼핑몰 회원의 주문 건수에 대한 통계치를 내는 과정이 OLAP 의 예시로 들 수 있을 것 같습니다.

**Comment by Irisation23** [comment]
<img width="761" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/53285909/a8e76f83-5c89-42ef-b4d1-37da941cf75e">
